authority/mgmt/api/middleware.go

```diff
@@ -0,0 +1,154 @@
+package api
+
+import (
+	"context"
+	"crypto/x509"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/smallstep/certificates/api"
+	"github.com/smallstep/certificates/authority/mgmt"
+	"github.com/smallstep/certificates/linkedca"
+	"go.step.sm/crypto/jose"
+)
+
+type nextHTTP = func(http.ResponseWriter, *http.Request)
+
+// requireAPIEnabled is a middleware that ensures the Administration API
+// is enabled before servicing requests.
+func (h *Handler) requireAPIEnabled(next nextHTTP) nextHTTP {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if h.db == nil {
+			api.WriteError(w, mgmt.NewError(mgmt.ErrorNotImplementedType,
+				"administration API not enabled"))
+			return
+		}
+		next(w, r)
+	}
+}
+
+func (h *Handler) authorizeToken(r *http.Request, token string) (*linkedca.Admin, error) {
+	jwt, err := jose.ParseSigned(token)
+	if err != nil {
+		return nil, mgmt.WrapError(mgmt.ErrorUnauthorizedType, err, "adminHandler.authorizeToken; error parsing x5c token")
+	}
+
+	verifiedChains, err := jwt.Headers[0].Certificates(x509.VerifyOptions{
+		Roots: h.rootPool,
+	})
```

Comment:
This actually verifies that the certificate has `x509.ExtKeyUsageServerAuth`. That all of our certs have by default, but it looks to me that requiring `x509.ExtKeyUsageClientAuth` (using `x509.VerifyOptions.KeyUsages`) is more appropriate.